From a7c1da09d3c75cce54f4cfe2386134f2f9b4c15e Mon Sep 17 00:00:00 2001
From: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Date: Tue, 4 Oct 2022 18:00:10 +0300
Subject: [PATCH 04/49] s32: ddr: Align ddr_utils.h to format generated by DDR
 Tool

The ddr_utils.h file is changed w.r.t. the new format generated by
DDR Tool, to minimize the required changes when any newer code will be
integrated.

Issue: ALB-9354
Upstream-Status: Pending 

Signed-off-by: Andrei Cherechesu <andrei.cherechesu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 include/drivers/nxp/s32/ddr/ddr_utils.h     | 23 ++++++++++++++++++++-
 include/drivers/nxp/s32/ddr/s32g/ddr_plat.h |  1 +
 2 files changed, 23 insertions(+), 1 deletion(-)

diff --git a/include/drivers/nxp/s32/ddr/ddr_utils.h b/include/drivers/nxp/s32/ddr/ddr_utils.h
index 52f994864..9731d85c8 100644
--- a/include/drivers/nxp/s32/ddr/ddr_utils.h
+++ b/include/drivers/nxp/s32/ddr/ddr_utils.h
@@ -31,7 +31,21 @@
 #ifndef DDR_UTILS_H_
 #define DDR_UTILS_H_
 
+#include <lib/mmio.h>
+
+#ifdef TRUSTED_BOARD_BOOT
 #include <ddr_plat.h>
+#else
+#include <stdbool.h>
+#include <stdlib.h>
+#endif
+
+/* Uncomment to store the CSR registers after executing DDR training */
+/* #define STORE_CSR_ENABLE */
+
+#ifndef dsb
+#define dsb()	__asm("DSB SY")
+#endif
 
 /* Possible errors */
 #define NO_ERR              0x00000000U
@@ -322,7 +336,14 @@
 #if !defined(PLAT_s32r)
 /* Standby SRAM */
 #define STNDBY_RAM_BASE           0x24000000
-#define RETENTION_ADDR            BL31SSRAM_CSR_BASE
+
+/*
+* This should be overwritten to store the configuration registers at different
+* address. Default one is the beginning of standby RAM.
+*/
+#ifndef RETENTION_ADDR
+#define RETENTION_ADDR            STNDBY_RAM_BASE
+#endif
 #endif
 
 /* DDR Subsystem */
diff --git a/include/drivers/nxp/s32/ddr/s32g/ddr_plat.h b/include/drivers/nxp/s32/ddr/s32g/ddr_plat.h
index 0ba1f53cb..035f2146e 100644
--- a/include/drivers/nxp/s32/ddr/s32g/ddr_plat.h
+++ b/include/drivers/nxp/s32/ddr/s32g/ddr_plat.h
@@ -11,5 +11,6 @@
 #include <plat/nxp/s32g/bl31_ssram/ssram_mailbox.h>
 
 #define STORE_CSR_ENABLE
+#define RETENTION_ADDR		BL31SSRAM_CSR_BASE
 
 #endif /* DDR_PLAT_H_ */
-- 
2.25.1

